components/suggest/src/error.rs

```diff
@@ -29,15 +32,35 @@ pub(crate) enum Error {
 #[derive(Debug, thiserror::Error)]
 #[non_exhaustive]
 pub enum SuggestApiError {
+    #[error("Network error: {reason}")]
+    NetworkError { reason: String },
+    // The application interrupted a request
+    #[error("Interrupted")]
+    Interrupted,
     #[error("Other error: {reason}")]
     Other { reason: String },
```

Comment:
If we go with `NetworkError`, let's maybe rename `Other` to `OtherError` or `UnexpectedError`, too?

Reply:
I ended up dropping the trailing `Error` from all variants.  I always wonder what the best way is, but at least this way is consistent.

Reply:
my 2c is that the redundant trailing `Error` is redundant :)

Reply:
Sold! 😆